### PR TITLE
Upgrade aws auth, fix CF template bug

### PR DESCRIPTION
**Pull request by flah00**
## Description 

* Fix bug in stack-template for controller, around `OpenICMP` and ssh security groups
* Upgrade to aws-iam-authenticator 0.5.1
* Resolve issue introduced in 941fd29, where the context cluster name didn't match
* Fix formatting issue with controller and worker kubeconfig files
* Add clusterrole, binding, and service account
* Allow aws-iam-auth to be scheduled on k8s 1.15 and 1.16

## Versions

* I have successfully created a new stack, on master
* I have not tested this on v0.16.x
* I have successfully created a new stack, on v0.15.x
    * This can be safely merged into v0.15.x after #1874 and #1886

## Installing the aws-iam-authenticator

AWS authenticator expects IAM roles for controllers and workers already exist. As it is currently configured, the aws-iam-authenticator must be enabled *after* the initial `kube-aws apply`. This is because the exports on the root stack, for workers, are not available to the control plane stack during creation, ie 

`+00:06:18       CREATE_FAILED                           test-Controlplane-13Y4STRG1AGDV        "No export named test-NodePoolPool1eWorkerIAMRoleArn found"`

### CODE REVIEW

**Comment by flah00** [comment]
Once the aws iam authenticator is enabled, the root stack can no longer be deleted, because the root stack exports a variable the control plane relies on. To delete the root stack you must first delete the control plane. 

```
Export k8s-NodePoolPool1eWorkerIAMRoleArn cannot be deleted as it is in use by k8s-Controlplane-1VKY701ZEUMW4
```

**Comment by dominicgunn** [comment]
Hey @flah00,

Thanks again for the work. Could you add a comment about that root stack deletion somewhere appropriate so that it doesn't just exist in this PR?

File: docs/getting-started/step-6-configure-add-ons.md [hunk 1]
```diff
@@ -119,6 +119,71 @@ Resources:
       ...
 ```
 
+## aws-iam-authenticator
```
Comment (flah00) [inline docs/getting-started/step-6-configure-add-ons.md:122]: @dominicgunn here's my first try at documenting this featuer
what do you tihnk?

Comment (dominicgunn) [reply docs/getting-started/step-6-configure-add-ons.md:122]: Sorry I took a while to get back to you. 

I've taken a glance and this is great & much better than our existing stuff. Thanks so much for the contribution.

File: docs/getting-started/step-6-configure-add-ons.md [hunk 2]
```diff
@@ -119,6 +119,71 @@ Resources:
...
+1. `kube-aws apply`
+
+### Delete an aws-iam-authenticator enabled cluster
+
+To successfully delete a cluster with aws-iam-authenticator enabled you must
+work around a quirk in cloud formation. You have two options.
+
+#### Delete the control plane stack manually and then delete the cluster
+
+1.  https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks?filteringText=&filteringStatus=active&viewNested=true&hideStacks=false
```
Comment (dominicgunn) [inline docs/getting-started/step-6-configure-add-ons.md:174]: I'd maybe wrap this in a tag, e.x.

```
1. [Open CloudFormation](https://console....)
```

File: builtin/files/stack-templates/control-plane.json.tmpl [hunk 3]
```diff
@@ -217,9 +217,11 @@
             "FromPort": -1,
             "IpProtocol": "icmp",
             "ToPort": -1
-          },
+          }
           {{end -}}
-          {{ range $_, $r := $.SSHAccessAllowedSourceCIDRs -}}
```
Comment (dominicgunn) [inline builtin/files/stack-templates/control-plane.json.tmpl:222]: Ack, we never ported this to the master branch? Awesome catch.

**Pull request merged**